index.js

```diff
@@ -172,30 +172,38 @@ var BrowserStackBrowser = function (
     url = url || previousUrl
     previousUrl = url
 
-    // TODO(vojta): handle non os/browser/version
-    var settings = {
-      os: args.os,
-      os_version: args.os_version,
-      device: args.device,
-      browser: args.browser,
-      tunnelIdentifier: bsConfig.tunnelIdentifier,
-      // TODO(vojta): remove "version" (only for B-C)
-      browser_version: args.browser_version || args.version || 'latest',
-      url: url + '?id=' + id,
-      'browserstack.tunnel': true,
-      timeout: bsConfig.timeout || 300,
-      project: args.project || bsConfig.project,
-      name: args.name || bsConfig.name || 'Karma test',
-      build: args.build || bsConfig.build ||
-        process.env.BUILD_NUMBER ||
+    var globalSettings = Object.assign(
+      {
+        timeout: 300,
+        name: 'Karma test',
+        build: process.env.BUILD_NUMBER ||
         process.env.BUILD_TAG ||
         process.env.CI_BUILD_NUMBER ||
         process.env.CI_BUILD_TAG ||
         process.env.TRAVIS_BUILD_NUMBER ||
         process.env.CIRCLE_BUILD_NUM ||
         process.env.DRONE_BUILD_NUMBER || null,
-      video: bsConfig.video !== undefined ? bsConfig.video : true
-    }
+        // TODO(vojta): remove "version" (only for B-C)
+        browser_version: args.version || 'latest',
+        video: true
+      },
+      bsConfig
+    )
+
+    var browserSpecificSettings = Object.assign(
+      {},
+      args
```

Comment:
Sorry for the nitpicky review, but since this is literally doing nothing, you can just pass the `args` to the `Object.assign` below.

Reply:
I've removed the `object.assign` completely, as the first argument is technically the target object, so we are no longer performing a clone. I do think the variable renaming makes the code more readable though.

Reply:
Yeah noticed that. I think that's fine.